python/neuroglancer/pipeline/storage.py

```diff
@@ -38,59 +43,75 @@ def __init__(self, layer_path='', n_threads=20):
         self._n_threads = n_threads
 
         if self._path.protocol == 'file':
-            self._interface = FileInterface
+            self._interface_cls = FileInterface
         elif self._path.protocol == 'gs':
-            self._interface = GoogleCloudStorageInterface
+            self._interface_cls = GoogleCloudStorageInterface
         elif self._path.protocol == 's3':
-            self._interface = S3Interface
+            self._interface_cls = S3Interface
+
+        self._interface = self._interface_cls(self._path)
```

Comment:
This is a particularly important fix, because this is is the cause of the Too Many files open we are seeing in production